File: EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/StatisticsUpdateHistories/StatisticsUpdateHistoryDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.StatisticsUpdateHistories;

public class StatisticsUpdateHistoryDto
{
    public long Id { get; set; }
    public long? StatisticsDraftId { get; set; }
    public long? SubmittedStatisticsId { get; set; }
    public long SchoolId { get; set; }
    public string ChangeType { get; set; } = string.Empty;
    public string ChangeCategory { get; set; } = string.Empty;
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public DateTime ChangeDate { get; set; }
    public string? UpdateReason { get; set; }
    public string? SupportingDocumentUrl { get; set; }
    public long? ChangedByUserId { get; set; }
    public bool IsApproved { get; set; }
    public long? ApprovedByUserId { get; set; }
    public DateTime? ApprovalDate { get; set; }
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}

File: src/app/modules/m6-StatisticsReports/data-access/models/statistics-update-history.dto.ts
import { BaseAuditFields, ChangeCategory, ChangeType } from './base.types';

export interface StatisticsUpdateHistory extends BaseAuditFields {
  statisticsDraftId: number | null;
  submittedStatisticsId: number | null;
  schoolId: number;
  changeType: ChangeType;
  changeCategory: ChangeCategory;
  oldValue: string | null;
  newValue: string | null;
  changeDate: string;
  updateReason: string | null;
  supportingDocumentUrl: string | null;
  changedByUserId: number | null;
  isApproved: boolean;
  approvedByUserId: number | null;
  approvalDate: string | null;
  notes: string | null;
}

export type CreateStatisticsUpdateHistory = Omit<StatisticsUpdateHistory, 'id' | 'createdAt' | 'modifiedAt'>;

export type UpdateStatisticsUpdateHistory = Pick<StatisticsUpdateHistory, 'id'> & Partial<Omit<StatisticsUpdateHistory, 'id' | 'createdAt' | 'modifiedAt'>>;
